### PR TITLE
CreatingNewVectorCpFunctions

### PR DESCRIPTION
**Pull request by andrewram4287**

### CODE REVIEW

**Comment by andrewram4287** [comment]
Currently the Cp factors are not the same, would you be able to check where I am going wrong either from Cp to vector or from vector to Cp?

File: gmm/figures/figure4.py [hunk 1]
```diff
@@ -29,7 +30,13 @@ def makeFigure():
     # conditions and output of decomposition
 
     rank = 2
-    _, _ = tensor_decomp(tMeans, rank, "NNparafac")
+    factors_NNP, facinfo = tensor_decomp(tMeans, rank, "NNparafac")
+
+    cpVector = cp_to_vector(facinfo, zflowTensor,maxcluster)
+
+    vectorFac = vector_to_cp(cpVector, rank, tMeans)
```
Comment (aarmey) [inline gmm/figures/figure4.py:36]: This test should go into `test_GMM.py`.

**Review by aarmey — CHANGES_REQUESTED** [review]

File: gmm/tensor.py [hunk 2]
```diff
@@ -49,6 +49,37 @@ def tensor_R2X(tensor, maxrank, tensortype):
 
     return rank, varexpl
 
+def cp_to_vector(facinfo,zflowTensor: xa.DataArray,maxcluster: int):
+
+    tensor = tl.cp_to_tensor(facinfo)
```
[the remark this answers is not in the document]
Comment (andrewram4287) [reply gmm/tensor.py:54]: That makes more sense! I misinterpreted your message before.

**Review by aarmey — APPROVED** [review]
Excellent. Merging!

You should now be able to fit the factors, instead of tMeans in full tensor form. Do you see how?

**Pull request merged**